File: front/lib/data/models/user_model/user_image.dart
class UserImage {
  String? imageNo;
  String? imageUrl;
  String? imageProfile;
  String? userNo;

  UserImage({
    this.imageNo,
    this.imageUrl,
    this.imageProfile,
    this.userNo,
  });

  Map<String, dynamic> toJson() {
    return {
      "imageNo": imageNo,
      "imageUrl": imageUrl,
      "imageProfile": imageProfile,
      "userNo": userNo,
    };
  }

  UserImage.fromJson(Map<String, dynamic> json)
      : imageNo = json['imageNo'],
        imageUrl = json['imageUrl'],
        imageProfile = json['imageProfile'],
        userNo = json['userNo'];

  UserImage copyWith({
    String? imageNo,
    String? imageUrl,
    String? imageProfile,
    String? userNo,
  }) {
    return UserImage(
      imageNo: imageNo ?? this.imageNo,
      imageUrl: imageUrl ?? this.imageUrl,
      imageProfile: imageProfile ?? this.imageProfile,
      userNo: userNo ?? this.userNo,
    );
  }

  @override
  String toString() {
    return 'UserImage{imageNo: $imageNo, imageUrl: $imageUrl, imageProfile: $imageProfile, userNo: $userNo}';
  }
}

File: back/src/main/java/com/pingo/entity/users/UserImage.java
package com.pingo.entity.users;

import com.pingo.exception.BusinessException;
import com.pingo.exception.ExceptionCode;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.ToString;

import java.util.UUID;

@Getter
@ToString
@AllArgsConstructor
@NoArgsConstructor
public class UserImage {
    private String imageNo;
    private String imageUrl;
    private String imageProfile;
    private String userNo;

    // imageNo 생성 로직 (중복 호출시 예외 처리)
    public String createImageNo() {
        if(this.imageNo == null) {
            String uuid = UUID.randomUUID().toString();
            return "UI" + uuid.substring(0, 8);
        }else {
            throw new BusinessException(ExceptionCode.DUPLICATE_IMAGE_NO);
        }
    }

    public void makeImageNo() {
        this.imageNo = createImageNo();
    }
}
